OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/AnalyticsProvider.java

```diff
@@ -0,0 +1,327 @@
+package org.edx.mobile.module.analytics;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.google.inject.Singleton;
+
+import org.edx.mobile.util.images.ShareUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An analytics provider class whose function is to contain all the analytics services we are using
+ * in our app, and call their respective APIs to send out analytics events.
+ */
+@Singleton
+public class AnalyticsProvider implements Analytics {
+    @NonNull
+    private List<Analytics> services = new ArrayList<>();
+
+    public void addAnalyticsProvider(@NonNull Analytics provider) {
+        services.add(provider);
+    }
+
+    public void trackScreenView(String screenName) {
+        trackScreenView(screenName, null, null);
+    }
+
+    public void trackScreenView(@NonNull String screenName, @Nullable String courseId,
```

Comment:
We might be able to get rid of this method of `trackScreenView` since we check for a null value anyway https://github.com/edx/edx-app-android/blob/khalid/MA-2981/OpenEdXMobile/src/main/java/org/edx/mobile/module/analytics/SegmentAnalytics.java#L202